src/Bitmap_cubical_complex/include/gudhi/Bitmap_cubical_complex_base.h

```diff
@@ -467,6 +495,128 @@ class Bitmap_cubical_complex_base {
 
+  //****************************************************************************************************************//
+  //****************************************************************************************************************//
+  //****************************************************************************************************************//
+  //****************************************************************************************************************//
+  class Vertices_iterator {
+    public:
+    Vertices_iterator(Bitmap_cubical_complex_base& b) : b(b) {
+      this->counter = std::vector<std::size_t>(b.dimension());
+    }
+
+    Vertices_iterator operator++() {
+      // first find first element of the counter that can be increased:
+      std::size_t dim = 0;
+      while ((dim != this->b.dimension()) && (this->counter[dim] == this->b.sizes[dim])) ++dim;
+
+      if (dim != this->b.dimension()) {
+        ++this->counter[dim];
+        for (std::size_t i = 0; i != dim; ++i) {
+          this->counter[i] = 0;
+        }
+      } else {
+        ++this->counter[0];
+      }
+      return *this;
+    }
+
+    Vertices_iterator operator++(int) {
+      Vertices_iterator result = *this;
+      ++(*this);
+      return result;
+    }
+
+    Vertices_iterator& operator=(const Vertices_iterator& rhs) {
+      this->counter = rhs.counter;
+      this->b = rhs.b;
+      return *this;
+    }
+
+    bool operator==(const Vertices_iterator& rhs) const {
+      if (&this->b != &rhs.b) return false;
+      if (this->counter.size() != rhs.counter.size()) return false;
+      for (std::size_t i = 0; i != this->counter.size(); ++i) {
+        if (this->counter[i] != rhs.counter[i]) return false;
+      }
+      return true;
+    }
+
+    bool operator!=(const Vertices_iterator& rhs) const { return !(*this == rhs); }
+
+    /*
+     * The operator * returns position of a cube in the structure of cubical complex. This position can be then used as
+     * an argument of the following functions:
+     * get_boundary_of_a_cell, get_coboundary_of_a_cell, get_dimension_of_a_cell to get information about the cell
+     * boundary and coboundary and dimension
+     * and in function get_cell_data to get a filtration of a cell.
+     */
+    std::size_t operator*() { return this->compute_index_in_bitmap(); }
+
+    std::size_t compute_index_in_bitmap() const {
+      std::size_t index = 0;
+      for (std::size_t i = 0; i != this->counter.size(); ++i) {
+        index += 2 * this->counter[i] * this->b.multipliers[i];
+      }
+      return index;
+    }
+
+    void print_counter() const {
+      for (std::size_t i = 0; i != this->counter.size(); ++i) {
+        std::clog << this->counter[i] << " ";
+      }
+    }
+    friend class Bitmap_cubical_complex_base;
+
+   protected:
+    std::vector<std::size_t> counter;
+    Bitmap_cubical_complex_base& b;
```

Comment:
I agree, but in this case and for the sake of homogeneity, all the classes attributes should have the same naming (`_ `at the end) and they are quite a lot (mainly in `Bitmap_cubical_complex.h` and `Bitmap_cubical_complex_base.h`)...
So, do we change everything? Only a few (occurrences of `b` and `counter`) (but then sometimes it will be inconsistent with other attributes such as `dimension` in `Skeleton_simplex_range`)? We don't change anything after all (at least in this PR)? (maybe leave it for a PR specific to correct naming in `Bitmap_cubical_complex` and maybe somewhere else (Open an issue for it))? (especially that this kind of modifications is prone to omissions/bugs...)